### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by husush23**
# The following have been addressed in this pull request:

### Logic

- Each class from the [UML class diagram](./images/catalog_of_my_things.png) should be created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
- All properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- All methods visible in the diagram should be implemented:
    - add_item method (in all classes that have that method)
        - should take an instance of the Item class as an input.
        - should add the input `item` to the collection of items.
        - should add self as a property of the item object (by using the correct setter from the item object).
    - can_be_archived?() in the `Item` class
        - should return true if published_date is older than 10 years.
        - otherwise, it should return false.
    - move_to_archive() in the `Item` class
        - should reuse can_be_archived?() method.
        - should change the archived property to true if the result of the can_be_archived?() method is true.
        - should do nothing if the result of the can_be_archived?() method is false.
    - can_be_archived?() in the `Book` class
        - should override the method from the parent class.
        - should return true if parent's method returns true OR if cover_state equals to "bad".
        - otherwise, it should return false.
    - can_be_archived?() in the `MusicAlbum` class
        - should override the method from the parent class.
        - should return true if parent's method returns true AND if on_spotify equals true.
        - otherwise, it should return false.
    - can_be_archived?() in the `Game` class
        - should override the method from the parent class.
        - should return true if parent's method returns true AND if last_played_at is older than 2 years.
        - otherwise, it should return false.
- Add unit tests for all implemented methods.

### User interface

- Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - If needed, ask for parameters for the option.
    - Have a way to quit the app.
- The following options should be available:
    - List all `books`
    - List all music albums
    - List of `games`
    - List all `genres` (e.g 'Comedy', 'Thriller')
    - List all `labels` (e.g. 'Gift', 'New')
    - List all `authors` (e.g. 'Stephen King')
    - List all `sources` (e.g. 'From a friend', 'Online shop')
    - Add a `book`
    - Add a music album
    - Add a `game`
 - All data should be preserved by saving collections in .json files.

### Database
- Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

### Project documentation

Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a **maximum length of 5 minutes**. The content of the video should include:

- A description of the project.
- A demo of the project features.
- You should also highlight some interesting piece of code or something you built that you are very proud of.
- You all should appear in the video and talk about the project.

### CODE REVIEW

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require_relative 'item'
+
+class Book < Item
```
Comment (muneebulrehman) [inline lib/book.rb:3]: As per the UML diagram provided to you, while creating a book, it should ask for `Genre`, `Author`, and `Label` details. However, in your case, it is only asking for the Label. Kindly fix this issue. A book belongs to an Item class and the item class has Genre, Author, and Label.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+class Game < Item
+  attr_accessor :multiplayer, :last_played_at, :author
+
+  def initialize(publish_date, multiplayer, last_played_at, id = nil)
+    super(publish_date, id)
+    @multiplayer = multiplayer
+    @last_played_at = last_played_at
+    @author = author
+  end
+
```
Comment (muneebulrehman) [inline lib/game.rb:13]: As per the UML class diagram, while creating a game, it should ask for `Genre`, `Author`, and `Label` details. However, in your case, it is only asking for the Author. Kindly fix this issue.

File: lib/music_album.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require_relative 'item'
+
+class MusicAlbum < Item
```
Comment (muneebulrehman) [inline lib/music_album.rb:3]: s per the UML class diagram, while creating a music album, it should ask for `Genre`, `Author`, and `Label` details. However, in your case, it is only asking for the Genre. Kindly fix this issue.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

_Note:_ If you have any questions, kindly ping me on slack. I go by the same name as on GitHub.

## Highlights
- No linter errors
- Descriptive PR
- All tests passing
- Good docs
- Good code structure.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shella12 — APPROVED** [review]
Hi @husush23, @Peter-Dumbari  and @Katended, 


## Status : APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Added genre, author and label for requested classes✔️ 
- Working console app 👍 
- Implemented all requirements 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**